frontend/src/scenes/actions/ActionsTable.tsx

```diff
@@ -46,8 +46,39 @@ export function ActionsTable(): JSX.Element {
     const { loadActions } = useActions(actionsModel)
     const [searchTerm, setSearchTerm] = useState('')
     const [filterByMe, setFilterByMe] = useState(false)
+    const [dupVisible, setDupVisible] = useState(false)
+    const [dupName, setDupName] = useState('')
+    const [dupAction, setDupAction] = useState<ActionType>()
     const { user, hasAvailableFeature } = useValues(userLogic)
 
+    const duplicateAction = async (action: ActionType | undefined): Promise<void> => {
+        if (action) {
+            try {
+                const { id, action_id, ...partialAction } = action
+                const newActionSteps: ActionStepType[] | undefined = action.steps?.map(({ id, ...partialStep }) => ({
+                    ...partialStep,
+                }))
+                await api.actions.create({
+                    ...partialAction,
+                    name: dupName,
+                    steps: newActionSteps,
+                })
+                loadActions()
+                lemonToast.success('Action duplicated')
+            } catch (response: any) {
+                if (response.type === 'validation_error' && response.code === 'unique') {
```

Comment:
The empty block looks a bit weird to me. How about you make this case an empty return and move the code in the else block outside?